### DISCUSSION TITLE
KEDA ScaledObject Failed to ensure HPA is correctly created for ScaledObject Error

### DISCUSSION THREAD

**Discussion by MikeyCSharpDude** [kedacore/keda#4074] [Q&A / Need Help] [answered]
Hi, I am new to K8s, in general. I have installed the TriggerAuthentication object, successfully, but get an error when applying the YAML for the ScaledObject object, whose source is an Azure Queue. The error returned is "Failed to ensure HPA is correctly created for ScaledObject." 

Can anyone help as to why I am getting the aforementioned error and how to resolve it? This is blocking me from setting up KEDA for our AKS (Azure Kubernetes Service) Cluster. Thanks!

The HPA is enabled; here's its YAML:

> apiVersion: autoscaling/v2
> kind: HorizontalPodAutoscaler
> metadata:
>   creationTimestamp: "2023-01-05T15:59:23Z"
>   managedFields:
>   \- apiVersion: autoscaling/v1
>     fieldsType: FieldsV1
>     fieldsV1:
>       f:spec:
>         f:maxReplicas: {}
>         f:minReplicas: {}
>         f:scaleTargetRef: {}
>         f:targetCPUUtilizationPercentage: {}
>     manager: kubectl-autoscale
>     operation: Update
>     time: "2023-01-05T15:59:23Z"
>   \- apiVersion: autoscaling/v2
>     fieldsType: FieldsV1
>     fieldsV1:
>       f:status:
>         f:conditions:
>           .: {}
>           k:{"type":"AbleToScale"}:
>             .: {}
>             f:lastTransitionTime: {}
>             f:message: {}
>             f:reason: {}
>             f:status: {}
>             f:type: {}
>           k:{"type":"ScalingActive"}:
>             .: {}
>             f:lastTransitionTime: {}
>             f:message: {}
>             f:reason: {}
>             f:status: {}
>             f:type: {}
>           k:{"type":"ScalingLimited"}:
>             .: {}
>             f:lastTransitionTime: {}
>             f:message: {}
>             f:reason: {}
>             f:status: {}
>             f:type: {}
>         f:currentMetrics: {}
>         f:currentReplicas: {}
>         f:desiredReplicas: {}
>     manager: kube-controller-manager
>     operation: Update
>     subresource: status
>     time: "2023-01-05T15:59:38Z"
>   name: azure-vote-front-hpa
>   namespace: azure-vote
>   resourceVersion: "29421826"
>   uid: 5036c38f-4adb-4d3b-814f-466b4ec434f1
> spec:
>   maxReplicas: 10
>   metrics:
>   \- resource:
>       name: cpu
>       target:
>         averageUtilization: 50
>         type: Utilization
>     type: Resource
>   minReplicas: 1
>   scaleTargetRef:
>     apiVersion: apps/v1
>     kind: Deployment
>     name: azure-vote-front
> status:
>   conditions:
>   \- lastTransitionTime: "2023-01-05T15:59:38Z"
>     message: recommended size matches current size
>     reason: ReadyForNewScale
>     status: "True"
>     type: AbleToScale
>   \- lastTransitionTime: "2023-01-05T15:59:38Z"
>     message: the HPA was able to successfully calculate a replica count from cpu resource
>       utilization (percentage of request)
>     reason: ValidMetricFound
>     status: "True"
>     type: ScalingActive
>   \- lastTransitionTime: "2023-01-05T15:59:38Z"
>     message: the desired count is within the acceptable range
>     reason: DesiredWithinRange
>     status: "False"
>     type: ScalingLimited
>   currentMetrics:
>   \- resource:
>       current:
>         averageUtilization: 1
>         averageValue: 1m
>       name: cpu
>     type: Resource
>   currentReplicas: 1
>   desiredReplicas: 1

Here's the current KEDA ScaledObject Object YAML:

> apiVersion: keda.sh/v1alpha1
> kind: ScaledObject
> metadata:
>   creationTimestamp: "2023-01-05T18:23:12Z"
>   finalizers:
>   \- finalizer.keda.sh
>   generation: 1
>   labels:
>     scaledobject.keda.sh/name: azure-queue-scaledobject
>   managedFields:
>   \- apiVersion: keda.sh/v1alpha1
>     fieldsType: FieldsV1
>     fieldsV1:
>       f:metadata:
>         f:finalizers:
>           .: {}
>           v:"finalizer.keda.sh": {}
>         f:labels:
>           .: {}
>           f:scaledobject.keda.sh/name: {}
>     manager: keda
>     operation: Update
>     time: "2023-01-05T18:23:12Z"
>   \- apiVersion: keda.sh/v1alpha1
>     fieldsType: FieldsV1
>     fieldsV1:
>       f:status:
>         .: {}
>         f:conditions: {}
>         f:originalReplicaCount: {}
>         f:scaleTargetGVKR:
>           .: {}
>           f:group: {}
>           f:kind: {}
>           f:resource: {}
>           f:version: {}
>         f:scaleTargetKind: {}
>     manager: keda
>     operation: Update
>     subresource: status
>     time: "2023-01-05T18:23:12Z"
>   \- apiVersion: keda.sh/v1alpha1
>     fieldsType: FieldsV1
>     fieldsV1:
>       f:spec:
>         .: {}
>         f:maxReplicaCount: {}
>         f:minReplicaCount: {}
>         f:scaleTargetRef:
>           .: {}
>           f:apiVersion: {}
>           f:kind: {}
>           f:name: {}
>         f:triggers: {}
>     manager: rancher
>     operation: Update
>     time: "2023-01-05T18:23:12Z"
>   name: azure-queue-scaledobject
>   namespace: azure-vote
>   resourceVersion: "29496628"
>   uid: ec30e8d5-9f58-41d3-bdb9-2af3372d683c
> spec:
>   maxReplicaCount: 10
>   minReplicaCount: 1
>   scaleTargetRef:
>     apiVersion: apps/v1
>     kind: Deployment
>     name: azure-vote-front
>   triggers:
>   \- metadata:
>       accountName: pubsharestoragedev
>       cloud: Private
>       connectionFromEnv: AzureWebJobsStorage
>       endpointSuffix: queue.local.azurestack.external
>       queueLength: "1"
>       queueName: pubshare-keda-dev
>     type: azure-queue
> status:
>   conditions:
>   \- message: Failed to ensure HPA is correctly created for ScaledObject
>     reason: ScaledObjectCheckFailed
>     status: "False"
>     type: Ready
>   \- message: ScaledObject check failed
>     reason: UnkownState
>     status: Unknown
>     type: Active
>   \- status: Unknown
>     type: Fallback
>   originalReplicaCount: 1
>   scaleTargetGVKR:
>     group: apps
>     kind: Deployment
>     resource: deployments
>     version: v1
>   scaleTargetKind: apps/v1.Deployment

**Comment by zroubalik** [comment, marked as answer]
Could you please look into KEDA Operator logs for details?

**Reply by MikeyCSharpDude** [reply]
Thanks, as this pointed me in the right direction: it's that it can't get the Azure Storage Account connection string: 1.6729592500766408e+09	ERROR	scalehandler	error resolving auth params	{"scalerIndex": 0, "object": {"apiVersion": "keda.sh/v1alpha1", "kind": "ScaledObject", "namespace": "azure-vote", "name": "azure-queue-scaledobject"}, "error": "**_error parsing azure queue metadata: no connection setting given_**"}

NOW, I wish it'd actually display this error, as opposed to it saying that the HPA is not correctly created.

**Reply by tomkerkhove** [reply]
With "display", what do you mean with that?

**Reply by tomkerkhove** [reply]
@zroubalik This may be something for https://github.com/kedacore/keda/issues/3764

**Reply by zroubalik** [reply]
yes, definitely!
